kube-core/src/params.rs

```diff
@@ -140,6 +140,39 @@ impl ListParams {
     }
 }
 
+/// The validation directive to use for `fieldValidation` when using server-side apply.
+#[derive(Clone, Debug)]
+pub enum ValidationDirective {
```

Comment:
As is, I think this probably should to be re-exported from `kube-client::api` (in kube-client/src/api/mod.rs) since we currently do it for all other param structs in there. It would be a bit annoying to miss out bits of the params even if that namespace is getting pretty overloaded.

Reply:
Good point

Reply:
Done, though I've also done the refactor to move to explicit methods for each validation and removed public visibility of `field_validation()`, meaning the enum never gets exposed outside of `PatchParams` at this point - should we change visibility of the entire enum?

Reply:
I think it's good to expose everything related to these structs. People can still use regular struct ctors and sometimes that's sensible in convoluted code.

I might even go so far as to keep `field_validation` as public as well. Maybe rename it to `::validation(vd)` so they are close together in documentation.

Reply:
Fair, will update tomorrow

Reply:
The risk of course, is that there's another validation thing being added to `PatchOptions` in the future, and then we'll have to prefix with `field_` on all the methods (via copy and deprecating old) to disambiguate, but I feel that's an OK choice.

Reply:
Kept the helpers and made the base one public